python/ray/dashboard/modules/train/train_head.py

```diff
@@ -95,43 +91,85 @@ async def get_train_runs(self, req: Request) -> Response:
         )
 
     async def _add_actor_status_and_update_run_status(self, train_runs):
-        from ray.train._internal.state.schema import ActorStatusEnum, RunStatusEnum
+        from ray.train._internal.state.schema import (
+            ActorStatusEnum,
+            RunStatusEnum,
+            TrainRunInfoWithDetails,
+            TrainWorkerInfoWithDetails,
+        )
 
-        actor_status_table = {}
         try:
             logger.info("Getting all actor info from GCS.")
-            request = gcs_service_pb2.GetAllActorInfoRequest()
-            reply = await self._gcs_actor_info_stub.GetAllActorInfo(request, timeout=5)
-            if reply.status.code == 0:
-                for message in reply.actor_table_data:
-                    actor_table_data = actor_table_data_to_dict(message)
-                    actor_status_table[actor_table_data["actorId"]] = actor_table_data[
-                        "state"
-                    ]
+            actors = await DataOrganizer.get_all_actors()
+
         except Exception:
             logger.exception("Error Getting all actor info from GCS.")
 
+        train_runs_with_details: List[TrainRunInfoWithDetails] = []
+
         for train_run in train_runs.values():
```

Comment:
Probably we can skip updating the hardware information for the finished train_runs to make it more efficient.

Reply:
good idea

Reply:
I decided not to do this since the amount of work is neglible

Reply:
oh yes. That makes sense.